tests/test_misc.py

```diff
@@ -514,13 +541,18 @@ def test_dict_to_file_csv_exception(mock_output_manager: OutputManager) -> None:
 
 
 def test_save_variables_to_csv_files_exceptions(
-    mock_output_manager: OutputManager
+    mock_output_manager: OutputManager,
 ) -> None:
     """Unit test for the function _save_variables_to_csv_files() in the file output_manager.py"""
-    invalid_path = '/invalid/directory'
 
-    with pytest.raises((PermissionError, OSError)):
-        mock_output_manager._save_variables_to_csv_files({}, 'filter', invalid_path)
+    with patch("pathlib.Path.mkdir") as mock_mkdir:
+        mock_mkdir.side_effect = OSError("test OS Error")
+        with pytest.raises(OSError, match="test OS Error"):
+            mock_output_manager._save_variables_to_csv_files({}, "filter", "path")
+
+        mock_mkdir.side_effect = FileNotFoundError("test File not Found Error")
+        with pytest.raises(OSError, match="test File not Found Error"):
```

Comment:
```suggestion
        with pytest.raises(FileNotFoundError, match="test File not Found Error"):
```
Makes the test a little more precise.